core/src/Borrower.sol

```diff
@@ -185,6 +229,57 @@ contract Borrower is IUniswapV3MintCallback {
         if (amount1 > 0) TOKEN1.safeTransfer(msg.sender, amount1);
     }
 
+    function _getAssetsAndWithdrawPositions(
+        int24[] memory positions_,
+        Prices memory prices
+    ) private returns (Assets memory assets) {
+        assets.fixed0 = TOKEN0.balanceOf(address(this));
+        assets.fixed1 = TOKEN1.balanceOf(address(this));
+
+        uint256 count = positions_.length;
+        unchecked {
+            for (uint256 i; i < count; i += 2) {
+                uint160 L;
+                uint160 U;
+                uint128 liquidity;
+
+                {
+                    // Load lower and upper ticks from the `positions_` array
+                    int24 l = positions_[i];
+                    int24 u = positions_[i + 1];
+                    // Fetch amount of `liquidity` in the position
+                    (liquidity, , , , ) = UNISWAP_POOL.positions(keccak256(abi.encodePacked(address(this), l, u)));
+
+                    // Compute lower and upper sqrt ratios
+                    L = TickMath.getSqrtRatioAtTick(l);
+                    U = TickMath.getSqrtRatioAtTick(u);
+
+                    // TODO what happens if we attempt to withdraw a position that doesn't exist?
```

Comment:
Following the calls:
1. `burn` calls [`_modifyPosition`](https://github.com/Uniswap/v3-core/blob/main/contracts/UniswapV3Pool.sol#L528-L535)
2. `_modifyPostion` calls [`_updatePosition`](https://github.com/Uniswap/v3-core/blob/main/contracts/UniswapV3Pool.sol#L322-L328)
3. `_updatePosition` calls [`positions.get`](https://github.com/Uniswap/v3-core/blob/main/contracts/UniswapV3Pool.sol#L389) - under the hood this is essential just querying the `mapping`. If my understanding of `mapping` is correct, this should succeed and will return [default values (all zeroes)](https://docs.soliditylang.org/en/v0.4.21/types.html#mappings) for all fields of the `Position`. 
4. We then call [`position.update`](https://github.com/Uniswap/v3-core/blob/05c10bf6d547d6121622ac51c457f93775e1df09/contracts/UniswapV3Pool.sol#L449) using the `position` object returned in step 3. It'll revert [here](https://github.com/Uniswap/v3-core/blob/05c10bf6d547d6121622ac51c457f93775e1df09/contracts/libraries/Position.sol#L54).

Reply:
Wrote a quick test to confirm this, and you're 100% correct. Added a conditional to `continue` early if `liquidity == 0`.